magento/magento2
# Fix #21725 - Render price on category list with correct precision for current locale

### Description (*)

I recreated https://github.com/magento/magento2/pull/21829 here, which got accidentally closed, when I tried to re-create changes on top of 2.5-develop. I adjusted changes to latest comment from @sivaschenko - while I will work on fixing test that might still fail, would be great if you could verify if such approach is what you meant.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#21725

### Manual testing scenarios (*)

1. Default locale set should be for Oman (ar_OM it's not by default available in Magento, need to be added in \Magento\Framework\Locale\Config but didn't check all locales to verify where it also occurs), Oman Rial (OMR) selected as currency
2. Visit product listing page - see that the products have prices with precision 3.
3. Visit product details page - see that the products have prices with precision 3.
4. Visit all other places on frontend where price is displayed - precision should be same there.

### Questions or comments


### Contribution checklist (*)
